File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        cur = 0
        for i in range(len(prices)):
            if i == 0:
                cur = prices[i]
            else:
                if prices[i] > cur:
                    ans = ans + prices[i]-cur
                    cur = prices[i]
                else:
                    cur = prices[i]
        return ans

File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:

    int rec(vector<int>&dp, int cur, int n) {
        if(cur == n) return 1;
        if(cur > n) return 0;
        if(dp[cur] != -1) return dp[cur];
        int ans = rec(dp, cur + 1, n);
        ans += rec(dp, cur + 2, n);
        return dp[cur] = ans;
    }
    int climbStairs(int n) {
        vector<int>dp(n, -1);
        
        return rec(dp, 0, n);

    }
};
